Edit files in preparation for adding files to quantum docs (#38)
Qiskit/qiskit-addon-sqd-hpc: docs/compilation-flags.rst

Compilation flags
************************************

> This guide documents flags, such as ``#define``, that can be passed to the compiler when using this library.

Reviewer: ```suggestion
This guide describes the compiler flags supported by this library, including ``#define``-based macros.
```

> Disable exceptions

Reviewer: I don't understand the purpose of this change.  I am concerned that the rephrased version could read as a command.  And a similar phrase on line 14 was left untouched.

Author: I should have applied the same phrasing suggestions everywhere, but we always leave off the "how to" and just leave it implied.  Having "how to" becomes repetitive and just adds extra words for translation.  But you can keep "how to" throughout if you want. It's not hugely important to me.

Reviewer: Would you be open to "Disabling exceptions" ? (and likewise on the RTTI line)

These aren't things that most users will worry about, but the information is there for the users who need them, which is why I'd like to avoid the headers sounding like an imperative.

Author: "How to" would be better, since we dislike gerunds ( -ing  verbs) even more :D

Reviewer: Ok, in that case my preference is to stick with "How to" :slightly_smiling_face: 

```suggestion
How to disable exceptions
```

   cmake .. -DCMAKE_CXX_FLAGS="-fno-exceptions -DQKA_SQD_DISABLE_EXCEPTIONS=1 -DDOCTEST_CONFIG_NO_EXCEPTIONS_BUT_WITH_ALL_ASSERTS"
   make

Note that compiling this repository's test suite requires that you set ``#define DOCTEST_CONFIG_NO_EXCEPTIONS_BUT_WITH_ALL_ASSERTS``, as exceptions must be disabled in the doctest framework and the SQD library.

Concepts (C++20 and later)
--------------------------

> This library is designed to use C++ concepts if they are available (for example, if the code is compiled according to C++20 or a later version of the standard).  If possible, users are encouraged to use such a compiler when developing, as this will likely lead to better error messages from the compiler (such as when a template argument has an unexpected type).

Reviewer: ```suggestion
This library is designed to use C++ concepts if they are available (that is, if the code is compiled according to C++20 or a later version of the standard).  If possible, users are encouraged to use such a compiler when developing, as this will likely lead to better error messages from the compiler (such as when a template argument has an unexpected type).
```
